app/models/course/condition/assessment.rb

```diff
@@ -0,0 +1,39 @@
+class Course::Condition::Assessment < ActiveRecord::Base
+  include ActiveSupport::NumberHelper
+
+  acts_as_condition
+  belongs_to :assessment, class_name: Course::Assessment.name, inverse_of: false
+
+  default_scope { includes(:assessment) }
+
+  def title
+    if minimum_grade_percentage
+      minimum_grade_percentage_display = number_to_percentage(minimum_grade_percentage,
+                                                              precision: 2,
+                                                              strip_insignificant_zeros: true)
+      self.class.human_attribute_name('title.title',
+                                      assessment_title: assessment.title,
+                                      minimum_grade_percentage: minimum_grade_percentage_display)
+    else
+      assessment.title
+    end
+  end
+
+  def satisfied_by?(course_user)
+    graded_submissions = graded_submissions_by_user(course_user.user)
+
+    if minimum_grade_percentage
+      graded_submissions.any? do |submission|
```

Comment:
would it be easier to define an attribute grade_percentage on the submission instead?